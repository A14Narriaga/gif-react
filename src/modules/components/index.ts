export * from "./card"
export * from "./card-grid"
export * from "./input"
