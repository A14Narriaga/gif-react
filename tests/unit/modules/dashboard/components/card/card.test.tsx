import { render, screen } from "@testing-library/react"
import { describe, expect, test } from "vitest"

import { Card } from "@src/modules"

describe("Tests in <Card.tsx />", () => {
	const gif = {
		title: "Test Gif",
		imgUrl: "https://example.com/test.gif"
	}

	test("Shoud match with snapshot", () => {
		const { container } = render(<Card {...gif} />)
		expect(container).toMatchSnapshot()
	})

	test("Should show the image with the correct url and alt", () => {
		render(<Card {...gif} />)
		// screen.debug()
		const { alt, src } = screen.getByRole<HTMLImageElement>("img")
		expect(alt).toBe(gif.title)
		expect(src).toBe(gif.imgUrl)
	})

	test("Should show the title", () => {
		render(<Card {...gif} />)
		const element = screen.getByText(gif.title)
		expect(element).toBeTruthy()
	})
})
