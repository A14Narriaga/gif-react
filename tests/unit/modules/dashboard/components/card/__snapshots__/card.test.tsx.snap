// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <Card.tsx /> > Shoud match with snapshot 1`] = `
<div>
  <div
    class="bg-white border border-gray-200 rounded-lg shadow dark:bg-gray-800 dark:border-gray-700"
  >
    <img
      alt="Test Gif"
      class="rounded-t-lg"
      src="https://example.com/test.gif"
    />
    <div
      class="p-5"
    >
      <p
        class="font-normal text-gray-700 dark:text-gray-400 truncate"
      >
        Test Gif
      </p>
    </div>
  </div>
</div>
`;
